sympy/sets/sets.py

```diff
@@ -1438,6 +1438,8 @@ def reduce(args):
                              if all(other.contains(x) == True for other in other_args)])
                 unk = [x for x in s
                        if any(other.contains(x) not in (True, False) for other in other_args)]
+                other_args = [elm - FiniteSet(*unk) if elm.is_FiniteSet and
+                                not FiniteSet(*unk).is_EmptySet else elm for elm in other_args]
```

Comment:
Under which conditions is ``FiniteSet(*unk).is_EmptySet`` true? Is it not equivalent to ``unk`` being empty?